internal/util/util.go

```diff
@@ -60,15 +61,29 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	switch opts.Type {
+	case define.BuildOutputImage:
+		if !opts.Push {
+			return nil
+		}
+
+		libimageOptions := &libimage.PushOptions{}
+		libimageOptions.Writer = os.Stdout
+		runtime, err := libimage.RuntimeFromStore(store, &libimage.RuntimeOptions{SystemContext: &types.SystemContext{}})
+		dest := fmt.Sprintf("%s:%s", opts.ImageRef.Transport().Name(), opts.ImageRef.StringWithinTransport())
+		_, err = runtime.Push(context.Background(), opts.Image, dest, libimageOptions)
+		if err != nil {
+			return fmt.Errorf("failed while pushing image %+q: %w", opts.ImageRef, err)
+		}
+	case define.BuildOutputLocal:
```

Comment:
This replaces an `opts.IsDir` condition … but the parser never sets `BuildOutputLocal` AFAICS.

I’d expect `IsDir` to be removed entirely, and replaced by the new `Type`.